### PR TITLE
feat: Add java sample for managed autoscaler

### PR DESCRIPTION
**Pull request by claire921**
Add java sample for managed autoscaler.

See the support for autoscaler here: https://github.com/claire921/java-spanner/pull/1/commits/e5e6923a351670ab237c411bb4a549533dac1b6b.

### CODE REVIEW

File: samples/snippets/src/main/java/com/example/spanner/CreateInstanceWithAutoscalingConfigExample.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2021 Google LLC
```
Comment (arpan14) [inline samples/snippets/src/main/java/com/example/spanner/CreateInstanceWithAutoscalingConfigExample.java:2]:
```suggestion
 * Copyright 2023 Google LLC
```

File: samples/snippets/src/main/java/com/example/spanner/CreateInstanceWithAutoscalingConfigExample.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    String instanceId = "my-instance";
+    createInstance(projectId, instanceId);
+  }
+
+  static void createInstance(String projectId, String instanceId) {
+    Spanner spanner = SpannerOptions.newBuilder().setProjectId(projectId).build().getService();
+    InstanceAdminClient instanceAdminClient = spanner.getInstanceAdminClient();
+
+    // Set Instance configuration.
+    String configId = "regional-us-central1";
```
Comment (arpan14) [inline samples/snippets/src/main/java/com/example/spanner/CreateInstanceWithAutoscalingConfigExample.java:47]: Why is config hard-coded?

File: samples/snippets/src/test/java/com/example/spanner/CreateInstanceWithAutoscalingConfigSampleIT.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+package com.example.spanner;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.cloud.spanner.InstanceId;
+import java.util.UUID;
+import org.junit.Test;
+
+public class CreateInstanceWithAutoscalingConfigSampleIT extends SampleTestBase {
+  @Test
```
Comment (arpan14) [inline samples/snippets/src/test/java/com/example/spanner/CreateInstanceWithAutoscalingConfigSampleIT.java:26]: Nit: add a new line

File: samples/snippets/src/test/java/com/example/spanner/CreateInstanceWithAutoscalingConfigSampleIT.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2022 Google LLC
```
Comment (arpan14) [inline samples/snippets/src/test/java/com/example/spanner/CreateInstanceWithAutoscalingConfigSampleIT.java:2]:
```suggestion
 * Copyright 2023 Google LLC
```

File: samples/snippets/src/main/java/com/example/spanner/CreateInstanceWithAutoscalingConfigExample.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    AutoscalingConfig autoscalingConfig =
+        AutoscalingConfig.newBuilder()
+            .setAutoscalingLimits(
+                AutoscalingConfig.AutoscalingLimits.newBuilder().setMinNodes(1).setMaxNodes(2))
+            .setAutoscalingTargets(
+                AutoscalingConfig.AutoscalingTargets.newBuilder()
+                    .setHighPriorityCpuUtilizationPercent(65)
+                    .setStorageUtilizationPercent(95))
+            .build();
+    String displayName = "Descriptive name";
```
Comment (arpan14) [inline samples/snippets/src/main/java/com/example/spanner/CreateInstanceWithAutoscalingConfigExample.java:58]: We don't need a variable declaration here. Can use the value inline at `setDisplayName`

File: samples/snippets/src/main/java/com/example/spanner/CreateInstanceWithAutoscalingConfigExample.java (hunk 2 above)
Comment (claire921) [reply samples/snippets/src/main/java/com/example/spanner/CreateInstanceWithAutoscalingConfigExample.java:47]: That seems to be the pattern for the create instance sample: https://github.com/googleapis/java-spanner/blob/main/samples/snippets/src/main/java/com/example/spanner/CreateInstanceExample.java#L40-L47 ? Is there a better way?

File: samples/snippets/src/main/java/com/example/spanner/CreateInstanceWithAutoscalingConfigExample.java (hunk 5 above)
Comment (claire921) [reply samples/snippets/src/main/java/com/example/spanner/CreateInstanceWithAutoscalingConfigExample.java:58]: That seems to be the way for create instance sample: https://github.com/googleapis/java-spanner/blob/main/samples/snippets/src/main/java/com/example/spanner/CreateInstanceExample.java#L40-L47 ? I'm ok with inlining it, just wonder if we need to be consistent for other createinstance samples?

File: samples/snippets/src/test/java/com/example/spanner/CreateInstanceWithAutoscalingConfigSampleIT.java (hunk 3 above)
Comment (claire921) [reply samples/snippets/src/test/java/com/example/spanner/CreateInstanceWithAutoscalingConfigSampleIT.java:27]: Done

File: samples/snippets/src/main/java/com/example/spanner/CreateInstanceWithAutoscalingConfigExample.java (hunk 2 above)
Comment (arpan14) [reply samples/snippets/src/main/java/com/example/spanner/CreateInstanceWithAutoscalingConfigExample.java:47]: Sure, we can follow existing convention.

File: samples/snippets/src/main/java/com/example/spanner/CreateInstanceWithAutoscalingConfigExample.java (hunk 5 above)
Comment (claire921) [reply samples/snippets/src/main/java/com/example/spanner/CreateInstanceWithAutoscalingConfigExample.java]: Updated to inline the display name.

**Review by arpan14 — APPROVED** [review]

**Pull request merged**